independent-projects/tools/registry-descriptor/README.md

```diff
@@ -0,0 +1,105 @@
+# Quarkus Registry Descriptor
+
+This project provides a model and a parser API to handle the Quarkus extension registry.
+
+## Definitions
+### Repository
+A repository is a local directory (which can be cloned from a Git repository) with the following structure: 
+
+```bash
+.
+├── extensions
+│   └── jsf.json
+└── platforms.json
+```
+
+### Extensions
+For maintenance purposes, each extension is declared in its own file and has the following structure:
+
+```json
+{
+  "group-id": "org.apache.myfaces.core.extensions.quarkus",
+  "artifact-id": "quarkus-myfaces",
+  "releases": [
+    {
+      "version": "2.3-next",
+      "quarkus-core":  "1.3.1.Final"
+    },
+    {
+      "version": "2.4-next",
+      "quarkus-core": "1.3.2.Final"
+    }
+  ]
+}
+``` 
+
+The Quarkus extension MUST be released to a Maven repository. The descriptor states the GAV and the Quarkus version for each release (which is listed here for performance purposes - if not specified, the parser will attempt to resolve using the Maven Resolver API)
+
+
+### Platforms 
+
+Platforms are a set of extensions of a specific version and MUST exist as a BOM. Since the order is important, it is declared as a single JSON (ordered by priority - the preferred BOMs in the top)
+
+```json
+[
+  {
+    "group-id": "io.quarkus",
```

Comment:
this description does not seem to match registry.json.

It has "versions' which i'm not sure what is about ?

Reply:
I just noticed that too, I copied from Alexey's repository and forgot to change. 
I'm working on it, thanks for spotting it ;)